src/UrlShortenerServiceProvider.php

```diff
@@ -72,6 +75,14 @@ protected function registerMacros()
             return app('url.shortener')->shortenAsync(...$parameters);
         });
 
+        UrlGenerator::macro('shortenUsing', function (string $driver, ...$parameters) {
+            return app('url.shortener')->shortenUsing($driver, ...$parameters);
```

Comment:
We would still want to be able to use the `shortenUsing` with the Facade (and the singleton)
e.g. `app('url.shortener')->shortenUsing(...)`
If we remove the `shortenUsing` from the manager it will be called on the default driver instance.

Reply:
My only concern is Single Responsibility. In Laravel a Manager is a Factory with Proxy capabilities.

Facade:

> A facade is an object that serves as a front-facing interface **masking more complex underlying or structural code**

Factory:

> Creating objects **without having to specify the exact class** of the object that will be created

Proxy:

> A proxy is a wrapper or agent object that is being called by the client to **access the real serving object** behind the scenes.

Using the Manager as a Facade as well is odd for me. That's why I think that only the Facade should have the `shortenUsing` methods.

Sure my opinion is debatable.

Reply:
I think these methods should be moved to the facade, which is the convenience layer.